tests/annotations/tests.py

```diff
@@ -120,17 +120,62 @@ def test_basic_annotation(self):
         for book in books:
             self.assertEqual(book.is_book, 1)
 
+    def test_basic_alias(self):
+        books = Book.objects.alias(
+            is_book=Value(1, output_field=IntegerField())
+        )
+        self.assertNotIn('is_book', str(books.query))
+
+    def test_basic_alias_annotation(self):
+        books = Book.objects.alias(
+            is_book_alias=Value(1, output_field=IntegerField())
+        ).annotate(is_book=F('is_book_alias'))
+        self.assertNotIn('is_book_alias', str(books.query))
```

Comment:
Ditto I'd avoid asserting against the string representation of the query, and assert against the returned `.values()`.

Reply:
@charettes `.values()` changes quite a lot in a query generation. Is it safe to assert against it?

Reply:
I you don't want to use `.values()` you could simply use `not hasattr(book, 'is_book_alias`)`?